### PR TITLE
Link variant

### PR DESCRIPTION
**Pull request by dani-mp**
**Resolves**: 
Part of https://linear.app/prismic/issue/DT-2469/aauser-when-i-create-a-link-field-and-allow-variants-i-can-see-an.

### Checklist

- [ ] A comprehensive Linear ticket, providing sufficient context and details to facilitate the review of the PR, is linked to the PR.
- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: src/types/value/link.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export type EmptyLinkField<
 > = {
 	link_type: Type | string
 	text?: string
+	variant?: string
```
Comment (angeloashmore) [inline src/types/value/link.ts:28]: ❓ #ask: Does this imply someone can choose a link variant without providing a URL, similar to providing text without a URL? Just asking for my own knowledge; the code looks good.

File: src/types/value/contentRelationship.ts [hunk 2]
```diff
@@ -45,4 +45,5 @@ export interface FilledContentRelationshipField<
 	isBroken?: boolean
 	data?: DataInterface
 	text?: string
+	variant?: string
```
Comment (angeloashmore) [inline src/types/value/contentRelationship.ts:48]: ⚠️ #issue: From [this Slack message](https://prismic-team.slack.com/archives/C02L3FN3AJK/p1733937871356369?thread_ts=1733933751.211369&cid=C02L3FN3AJK), it seems we are only supporting `variant` on link fields, not content relationship and link to media fields.

File: src/types/value/linkToMedia.ts [hunk 3]
```diff
@@ -25,4 +25,5 @@ export interface FilledLinkToMediaField {
 	height?: string | null
 	width?: string | null
 	text?: string
+	variant?: string
```
Comment (angeloashmore) [inline src/types/value/linkToMedia.ts:28]: ⚠️ #issue: From [this Slack message](https://prismic-team.slack.com/archives/C02L3FN3AJK/p1733937871356369?thread_ts=1733933751.211369&cid=C02L3FN3AJK), it seems we are only supporting `variant` on link fields, not content relationship and link to media fields.

File: src/types/value/contentRelationship.ts (hunk 2 above)
Comment (angeloashmore) [reply src/types/value/contentRelationship.ts:48]: I wonder if we need to add the `variant` property in `src/types/value/link.ts` as an intersection. Something like this pseudo-code:

```ts
type LinkField = (WebLinkField | LinkToMediaField | ContentRelationshipField) & {
  variant?: string
}
```

That keeps `variant` out of link to media and content relationship fields.

**Review by angeloashmore — COMMENTED** [review]
Looks good so far! I left a few questions and suggestions.

File: src/types/value/link.ts (hunk 1 above)
Comment (dani-mp) [reply src/types/value/link.ts:28]: yes, same rationale

File: src/types/value/contentRelationship.ts (hunk 2 above)
Comment (dani-mp) [reply src/types/value/contentRelationship.ts:48]: first message: i think what we want is to not add this option to the link to media or content relationship concrete fields, but you can still create a _generic_ link without the select and then in page builder select one of the types

second message: i understood from a conversation with @xrutayisire the other day that one might want to use a concrete version of the field and it should already contain all properties, as they're all exported individually

**Pull request closed without merging**